flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBWriteBatchWrapper.java

```diff
@@ -49,22 +53,35 @@
 
 	private final int capacity;
 
-	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB) {
-		this(rocksDB, null, 500);
+	@Nonnegative
+	private final long batchSize;
+
+	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB, long writeBatchSize) {
+		this(rocksDB, null, 500, writeBatchSize);
 	}
 
 	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB, @Nullable WriteOptions options) {
-		this(rocksDB, options, 500);
+		this(rocksDB, options, 500, DEFAULT_BATCH_SIZE);
+	}
+
+	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB, @Nullable WriteOptions options, long batchSize) {
+		this(rocksDB, options, 500, batchSize);
 	}
 
-	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB, @Nullable WriteOptions options, int capacity) {
+	public RocksDBWriteBatchWrapper(@Nonnull RocksDB rocksDB, @Nullable WriteOptions options, int capacity, long batchSize) {
 		Preconditions.checkArgument(capacity >= MIN_CAPACITY && capacity <= MAX_CAPACITY,
 			"capacity should be between " + MIN_CAPACITY + " and " + MAX_CAPACITY);
+		Preconditions.checkArgument(batchSize >= 0, "Max batch size have to be no negative.");
```

Comment:
We can use `@Nonnegative` for all function parameters instead of `checkArgument`